### PR TITLE
Ts db fk connections

### PR DESCRIPTION
**Pull request by niklasld**
done

### CODE REVIEW

File: ad [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  Import_food_data[m
+  TS_db_fk_connections[m
+  US_Admin_dashboard[m
+  US_Set_daily_calorie_allowance[m
+  admin-reset-password[m
+  bug-fix-after-creating-user-first-login-dont-work[m
+* [32mbug_create_user_at_login_doesnt_work[m
+  create-calorie-accounting[m
+  create-calorie-accounting2[m
+  create-user-with-login[m
```
Comment (hasandk) [inline ad]: ..

File: src/main/java/com/base/site/controllers/RecipeController.java [hunk 2]
```diff
@@ -1,16 +1,145 @@
...
+        model.addAttribute("foodlist", foodlist);
+        model.addAttribute("pfoodlist", pfoodlist);
+        model.addAttribute("keyword", keyword);
+        model.addAttribute("recipeFood", new RecipeFood());
+
+        return ADD_FOOD_TO_RECIPE;
+    }
+//
+//    @GetMapping("/addFoodToRecipe/{foodId}/{recipeId}")
+//    public String addFoodToRecipe(@PathVariable("foodId")long foodId, @PathVariable("recipeId")long recipeId,Model model) {
```
Comment (hasandk) [inline src/main/java/com/base/site/controllers/RecipeController.java]: needed ?

File: src/main/java/com/base/site/repositories/RecipeFoodRepository.java [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
+@Repository("RecipeFoodRepository")
 public interface RecipeFoodRepository extends JpaRepository<RecipeFood, Long> {
+    //@Query("SELECT f FROM RecipeFood f WHERE f.recipe.id = f.recipe)
+    //public List<RecipeFood> findAllByRecipe(Recipe recipe);
```
Comment (hasandk) [inline src/main/java/com/base/site/repositories/RecipeFoodRepository.java]: needed ?

File: src/main/java/com/base/site/repositories/RecipeRepository.java [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
+import com.base.site.models.Users;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
-@Repository
+import java.util.List;
+
+@Repository("RecipeRepository")
 public interface RecipeRepository extends JpaRepository<Recipe, Long> {
+    //List<RecipeFood> findAllByUser(Users loggedInUser);
```
Comment (hasandk) [inline src/main/java/com/base/site/repositories/RecipeRepository.java]: needed ?

File: src/main/java/com/base/site/services/RecipeServiceImpl.java [hunk 5]
```diff
@@ -40,4 +43,15 @@ public void deleteById(Long id) {
...
+
+    @Override
+    public List<Recipe> findAllFkUser(Users loggedInUser) {
+        return recipeRepository.findAllByFkUser(loggedInUser);
+    }
+
+    /*
+    @Override
+    public List<RecipeFood> findAllByUser(Users loggedInUser) {
+        return recipeRepository.findAllByUser(loggedInUser);
```
Comment (hasandk) [inline src/main/java/com/base/site/services/RecipeServiceImpl.java]: needed ?

File: src/main/resources/templates/fragments/food_list/food_list.html [hunk 6]
```diff
@@ -0,0 +1,54 @@
+<div th:fragment="searchFilterHTML">
+    Filter: <input type="text" name="keyword" id="keyword" size="50" th:value="${keyword}" required />
+    &nbsp;
+    <input type="submit" value="Search" />
+    &nbsp;
+    <input type="button" value="Clear" id="btnClear" onclick="clearSearch()" />
+</div>
+
+<thead th:fragment="addFoodTableHeader" class="container">
```
Comment (mrtauern) [inline src/main/resources/templates/fragments/food_list/food_list.html:9]: Hvad er dette for et fragment?

**Review by hasandk — APPROVED** [review]

**Pull request merged**